addons/l10n_in/models/res_config_settings.py

```diff
@@ -9,6 +9,23 @@ class ResConfigSettings(models.TransientModel):
 
     show_module_l10n_in = fields.Boolean(compute='_compute_show_module_l10n_in')
     group_l10n_in_reseller = fields.Boolean(implied_group='l10n_in.group_l10n_in_reseller', string="Manage Reseller(E-Commerce)")
+    group_l10n_in_multi_gstin = fields.Boolean(string="Multi GSTIN", implied_group='l10n_in.group_l10n_in_multi_gstin')
+
+    def set_values(self):
+        super().set_values()
+        #set company it self as unit in exsiting records
+        if self.group_l10n_in_multi_gstin:
+            for company in self.env.user.company_ids:
+                company.partner_id.sudo().write({'l10n_in_gstin_company_id': company})
```

Comment:
no need to set value of l10n_in_gstin_company_id. so remove set_values function here

Reply:
ok 👍